src/google-classroom-extractor/tests/courses/test_courses.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from pathlib import Path
+from unittest.mock import patch
+import pandas as pd
+from sqlalchemy import create_engine
+from google_classroom_extractor.api.courses import request_all_courses_as_df
+
+DB_FILE = "tests/courses/test_courses.db"
+
+
+def dataframe_row_count(dataframe) -> int:
+    return dataframe.shape[0]
+
+
+def db_row_count(test_db) -> int:
+    return test_db.engine.execute("SELECT COUNT(rowid) from Courses").scalar()
+
+
+def db_ending_name(test_db) -> int:
+    return test_db.engine.execute(
+        "SELECT name from Courses WHERE rowid = (SELECT MAX(rowid) from Courses)"
+    ).scalar()
+
+
+def db_room_by_course_id(test_db, course_id) -> int:
+    return test_db.engine.execute(
+        f"SELECT room from Courses WHERE id = '{course_id}'"
+    ).scalar()
+
+
+@patch("google_classroom_extractor.api.courses.request_latest_courses_as_df")
+def test_overlap_removal(mock_latest_courses_df):
+    Path(DB_FILE).unlink(missing_ok=True)
+    test_db = create_engine(f"sqlite:///{DB_FILE}", echo=True)
```

Comment:
By "formal" definition, this is no longer a unit test - it is connecting to an external resource, and therefore is an integration test. Since sqlite is so easy to use, and we gain value from the integration test, I think I can live with this so long as the user isn't left waiting for the test to complete (which I very much doubt will occur).